starlette/middleware/base.py

```diff
@@ -222,3 +222,6 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             await send({"type": "http.response.body", "body": chunk, "more_body": True})
 
         await send({"type": "http.response.body", "body": b"", "more_body": False})
+
+        if self.background:
+            await self.background()
```

Comment:
For full backward compatibility add it to constructor also:

```
class _StreamingResponse(Response):
    def __init__(
        self,
        content: AsyncContentStream,
        status_code: int = 200,
        headers: typing.Mapping[str, str] | None = None,
        media_type: str | None = None,
        background: BackgroundTask | None = None,                   <<<<<<<<<<<<<
        info: typing.Mapping[str, typing.Any] | None = None,
    ) -> None:
        self.info = info
        self.body_iterator = content
        self.status_code = status_code
        self.media_type = media_type
        self.background = background                                           <<<<<<<<<<<<<
        self.init_headers(headers)
```

Reply:
This is because some middleware may be referring to that field. Or below will work also

```
self.background = None
```

Reply:
No users should be calling the constructor unless they do something like `type(response)(...)` which is not something we need to support.

Reply:
Agree, this aligns with the class being marked as private. Then please include
```
self.background = None
```

For full backward compatibility, any response object injected into `BaseHTTPMiddleware` based middleware needs to have the `background` field (default None)

Reply:
Ok I see your point now with the edit. Good point. Added.

@Kludex this is another reason to move the background task logic out of responses _and_ a good reason to rework our Response inheritance so that `Response` and `StreamingResponse` both inherit from a `BaseResponse` that has the API and initialization of common bits.